spiketoolkit/sorters/mountainsort4/mountainsort4.py

```diff
@@ -58,116 +54,46 @@ def set_params(self, **params):
         self.params = params
     
     def _setup_recording(self, recording, output_folder):
-        pass
-        # Not done
+        # alias to params
+        p = self.params
+        # Bandpass filter
+        if p['freq_min'] is not None:
+            recording = st.preprocessing.bandpass_filter(recording=recording, freq_min=p['freq_min'],
+                                                         freq_max=p['freq_max'])
+
+        # Whiten
+        if p['whiten']:
+            recording = st.preprocessing.whiten(recording=recording)
+
+        # Check location
+        if 'location' not in recording.getChannelPropertyNames():
+            for i, chan in enumerate(recording.getChannelIds()):
+                recording.setChannelProperty(chan, 'location', [0, i])
+
+        self.recording = recording
```

Comment:
In the new API there is no self.recording any more.

Reply:
I know but it was the only way to get MS4 in the new format